src/thread_task.rs

```diff
@@ -834,163 +835,151 @@ pub unsafe extern "C" fn rav1d_worker_task(data: *mut c_void) -> *mut c_void {
             continue 'outer;
         }
 
-        let mut found = false;
-        let mut f = 0 as *mut Rav1dFrameContext;
-        let mut t = 0 as *mut Rav1dTask;
-        let mut prev_t = 0 as *mut Rav1dTask;
-        if (*c).n_fc > 1 as c_uint {
-            // run init tasks second
-            'init_tasks: for i in 0..(*c).n_fc {
-                let first: c_uint = ::core::intrinsics::atomic_load_seqcst(&mut (*ttd).first);
-                f = &mut *((*c).fc).offset(first.wrapping_add(i).wrapping_rem((*c).n_fc) as isize)
-                    as *mut Rav1dFrameContext;
-                if (*f).task_thread.init_done.load(Ordering::SeqCst) != 0 {
-                    continue 'init_tasks;
-                }
-                t = (*f).task_thread.task_head;
-                if t.is_null() {
-                    continue 'init_tasks;
-                }
-                if (*t).type_0 as c_uint == RAV1D_TASK_TYPE_INIT as c_int as c_uint {
-                    found = true;
-                    break 'init_tasks;
-                }
-                if (*t).type_0 as c_uint == RAV1D_TASK_TYPE_INIT_CDF as c_int as c_uint {
-                    // XXX This can be a simple else, if adding tasks of both
-                    // passes at once (in dav1d_task_create_tile_sbrow).
-                    // Adding the tasks to the pending Q can result in a
-                    // thread merging them before setting init_done.
-                    // We will need to set init_done before adding to the
-                    // pending Q, so maybe return the tasks, set init_done,
-                    // and add to pending Q only then.
-                    let p1 = (if !((*f).in_cdf.progress).is_null() {
-                        ::core::intrinsics::atomic_load_seqcst((*f).in_cdf.progress)
+        let (f, t, mut prev_t) = 'found: {
+            if (*c).n_fc > 1 as c_uint {
+                // run init tasks second
+                'init_tasks: for i in 0..(*c).n_fc {
+                    let first: c_uint = ::core::intrinsics::atomic_load_seqcst(&mut (*ttd).first);
+                    let f = &mut *((*c).fc)
+                        .offset(first.wrapping_add(i).wrapping_rem((*c).n_fc) as isize);
+                    if f.task_thread.init_done.load(Ordering::SeqCst) != 0 {
+                        continue 'init_tasks;
+                    }
+                    let t = if let Some(t) = f.task_thread.task_head.as_mut() {
```

Comment:
```suggestion
                    let Some(t) = f.task_thread.task_head.as_mut() {
```

Reply:
Would need an extra indented block, and I wanted to preserve the C control flow here. I don't really think we should change that right now.

Reply:
This shouldn't change control flow or add indentation?  At least that's what I meant to suggest.  It's just using let else blocks to simplify things.

Reply:
Sorry, my bad, I didn't understand this new let ... else syntax. Tried it but I didn't get it right. Huh. Did that now.